libs/trpc/src/lib/middleware/authenticatedProcedure.ts

```diff
@@ -0,0 +1,17 @@
+import { publicProcedure } from '../trpc';
+import { TRPCError } from '@trpc/server';
+
+export const authenticatedProcedure = publicProcedure.use((opts) => {
+  const session = opts.ctx.session;
+  if (!session) {
+    throw new TRPCError({
+      message: 'User must be logged in with a valid session',
```

Comment:
Could we swap this error to be more informative - looking for something like `You must pass a valid token as "Bearer TOKEN" via the Authorization header to access this procedure`.